shard.go

```diff
@@ -25,40 +25,52 @@ type cacheShard struct {
 	stats Stats
 }
 
-func (s *cacheShard) get(key string, hashedKey uint64) ([]byte, error) {
-	s.lock.RLock()
+func (s *cacheShard) get(key string, hashedKey uint64, skipLock bool) ([]byte, error) {
```

Comment:
Ah, and the same regarding `set` method 😉

Reply:
I am really missing parameter default values here (which would be false of cause to not change the API).

But I have no better idea. I could clone the method but this will result in a "lot" of duplicate code. I would basically implement the method again just leaving out the lock lines?

Since those calls are not public I think we can live with it. What do you think?

Reply:
There is no need for a default params. Both of the functions are from different worls - one is lock-free and another is not.

Combining them is a way to make an error quite easily.

That's why I suggest to split them into 2 different. Little code duplication isn't a problem (especially here).

Reply:
done, good? :)